icegatherer.go

```diff
@@ -308,46 +308,12 @@ func (g *ICEGatherer) collectStats(collector *statsReportCollector) {
 		for _, candidatePairStats := range agent.GetCandidatePairsStats() {
 			collector.Collecting()
 
-			state, err := toStatsICECandidatePairState(candidatePairStats.State)
+			stats, err := toICECandidatePairStats(candidatePairStats)
 			if err != nil {
 				g.log.Error(err.Error())
+				continue
```

Comment:
A slight change in behaviour here. Consolidate the conversion from `ice.CandidatePairStats` -> `webrtc.ICECandidatePairStats` in `stats.go` so that it can be use both here and in `getSelectedCandidatePairStats()` 👇 . In fact, linter flagged it for repeated code 🎉 .

Previously, code continued to add stats after just logging the error. Now, it does not. The consolidation code in `stats.go` returns an empty stats if there is an error here.

I think the state error should never happen. There is also a comment in code like
```
    default:                                                                                                                                                    
        // NOTE: this should never happen[tm]                                                                                                                   
        err := fmt.Errorf("%w: %s", errStatsICECandidateStateInvalid, state.String())                                                                           
        return StatsICECandidatePairState("Unknown"), err
```

So, just skipping adding the stat on error.